### PR TITLE
Example of simple i18n style.

### PR DESCRIPTION
**Pull request by headhunter45**
This is just a quick idea of what I'm proposing for translations. I don't think it will add much overhead other than having both sets of strings in the binary. I'd also add a new setting to track/set the current language.

### CODE REVIEW

**Comment by solosky** [comment]
Hi, thanks for your contribution.
Since this is a POC only PR, I would not going to merge this PR to main.

Futher more, would you please to enhance the PR and  have a comprehensive i18n feature? I would happy to merge the full featured i18n feature to main stream. 

Thanks.

**Comment by headhunter45** [comment]
I was hesitant to include any other libs. Can you give some examples of what you mean by more comprehensive. Would something like gettext or icu be ok?

**Comment by solosky** [comment]
sorry I make misunderstanding. 
From what I mean is that Pixl.js is a project built in my offwork time, so many features the users want to have, I would very happy to see a completed and fully tested feature PR waiting to merge main stream. 

As your PR is a POC like feature, would you please to commit more code to have this PR with  fully featured i18n function, includes:

* Fully English and Chinese UI translations
* A menu in settings page to switch UI language

To keep the project code simple, I do not recomend to introduce any libs. The simple way is to define a i18n string array that includes all supported language translation. 

For exmaple:

```
typedef struct {
    const uint16_t msg_id;
    const char*  en_str;
    const char*  zh_str;
} i18n_msg_t;


const i18n_msg_t m_i18n_msg_array [] = {
{I18N_MSG_APP_AMIIBO_EMULATOR, "Amiibo Emulator", "Amiibo模拟器"},
{I18N_MSG_APP_BLE_CONNECT, "BLE Connect", "蓝牙连接"},
...
}
```

**Comment by headhunter45** [comment]
Can you tell me what the Chinese for this would be "Language [%s]"?

**Comment by solosky** [comment]
> Can you tell me what the Chinese for this would be "Language [%s]"?

语言 [%s] 

Simplified Chinese: 简体中文
English: 英语

File: fw/application/src/i18n/messages.c [hunk 1]
```diff
@@ -0,0 +1,145 @@
+#include "messages.h"
+#include <stddef.h>
+
+static enum message_languages_t current_language = MESSAGE_LANG_ZH_CN;
+const enum message_languages_t default_language = MESSAGE_LANG_ZH_CN;
+
+const char* get_message_in_language(enum messages_t message_id, enum message_languages_t language_id) {
+    switch(language_id) {
+                case MESSAGE_LANG_ZH_CN:
```
Comment (solosky) [inline fw/application/src/i18n/messages.c:9]: please put the message to different c files with an array , for example:

message.h
```
enum message_id {
    MESSAGE_ID_AMIIBO_EMULATOR,
    MESSAGE_ID_BLE,
    MESSAGE_ID_SETTINGS,
    MESSAGE_ID_MAX
};
```
message.c
```
const char* get_message(message_id  id, message_languages_t lang){
   //TODO boundry check...
     if(lang == MESSAGE_LANG_ZH_CN) {
           return lang_zh_cn[id];
       } else {
           return  lang_en_us[id];
       }
}
```

lang_en_us.c
```
const char * lang_en_us[MESSAGE_ID_MAX] = {
    [MESSAGE_ID_AMIIBO_EMULATOR] = "Amiibo Emulator", 
    [MESSAGE_ID_SETTINGS] = "Settings"
};
```

lang_zh_cn.c
```
const char * lang_zh_cn[MESSAGE_ID_MAX] = {
    [MESSAGE_ID_AMIIBO_EMULATOR] = "Amiibo模拟器", 
    [MESSAGE_ID_SETTINGS] = "设置"
};
```

**Comment by solosky** [comment]
The firmware supports English officially since 2.7.0, close this PR since no more activities from author.

**Pull request closed without merging**